snapcraft/internal/sources/_base.py

```diff
@@ -0,0 +1,70 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import requests
+import shutil
+
+import snapcraft.internal.common
+from snapcraft.internal.indicators import (
+    download_requests_stream,
+    download_urllib_source
+)
+
+
+class IncompatibleOptionsError(Exception):
```

Comment:
can you put this in `snapcraft.sources.errors` and make it inherit from the base error class we have in `snapcraft.errors` (notice no `_`)